flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/listener/FlinkResourceListenerTest.java

```diff
@@ -58,36 +58,40 @@ public void testListeners() {
         var eventRecorder = EventRecorder.create(kubernetesClient, listeners);
 
         var deployment = TestUtils.buildApplicationCluster();
-        statusRecorder.updateStatusFromCache(deployment);
 
-        statusRecorder.patchAndCacheStatus(deployment);
         assertTrue(listener1.updates.isEmpty());
         assertTrue(listener2.updates.isEmpty());
         assertTrue(listener1.events.isEmpty());
         assertTrue(listener2.events.isEmpty());
 
-        deployment.getStatus().setJobManagerDeploymentStatus(JobManagerDeploymentStatus.ERROR);
-
-        statusRecorder.patchAndCacheStatus(deployment);
+        statusRecorder.updateStatusFromCache(deployment);
+        assertEquals(1, listener1.updates.size());
+        statusRecorder.updateStatusFromCache(deployment);
         assertEquals(1, listener1.updates.size());
         assertEquals(deployment, listener1.updates.get(0).getFlinkResource());
 
-        assertEquals(1, listener2.updates.size());
```

Comment:
Did we intentionally remove the checks against `listener2`?

Reply:
I'm comparing listener2 to listener1 later in the code in a for loop (line 73)

Reply:
... and obviously was my intention, but realised I made a mistake... fixing

Reply:
fixed it.